DataProcessing/datax-host/src/main/scala/datax/fs/HadoopClient.scala

```diff
@@ -319,9 +323,15 @@ object HadoopClient {
   def writeWithTimeoutAndRetries(hdfsPath: String,
                                  content: Array[Byte],
                                  timeout: Duration = Duration(5, TimeUnit.SECONDS),
-                                 retries: Int = 0
+                                 retries: Int = 0,
+                                 blobStorageKey: broadcast.Broadcast[String]
                                 ) = {
     val logger = LogManager.getLogger(s"FileWriter${SparkEnvVariables.getLoggerSuffix()}")
+    //Set hadoop config with storage account key on executer node
+	if(blobStorageKey != null){
+      var sa = getWasbStorageAccount(hdfsPath)
+      getConf().set(s"fs.azure.account.key.$sa${BlobProperties.BlobHostPath}", blobStorageKey.value)
```

Comment:
Could you just call function setStorageAccountKey instead of setting the key here so that key setting happens in only place?

Reply:
if we use setStorageAccountKey then job fails java.lang.NoSuchMethodError: com.microsoft.azure.storage.blob.CloudBlob.startCopyFromBlob error

Reply:
Extracted to another method to set hadoop conf